include/picongpu/fields/MaxwellSolver/ArbitraryOrderFDTD/Derivative.hpp

```diff
@@ -0,0 +1,262 @@
+     */
+     template<
+        uint32_t T_neighbors,
+        uint32_t T_direction
+     >
+     using ForwardDerivativeFunctor = detail::GeneralAofdtdDerivative< T_neighbors - 1, T_neighbors, T_neighbors, T_direction >;
+
+
+     /** Functor for backward derivative along the given direction used in ArbitraryOrderFDTD solver
+     *
+     * Compute an approximation of the derivative of a field f by a finite difference of
+     * order 2 * T_neighbors, where T_neighbors is the number of neighbors
+     * used to calculate the finite difference.
+     *
+     * This finite difference approximation for the derivative is computed on a staggered grid,
+     * i.e. the derivative will be known at a position i-1/2, if the field f is known
+     * at 2 * T_neighbors grid nodes i - T_neighbors, i - T_neighbors + 1, ..., i + T_neighbors - 1.
+
+     * The finite difference calculation can be expressed as a sum of finite differences where the
+     * distance of field components used in individual finite differences computations increases
+     *     D_x f(i-1/2) = sum_{l=0}^{T_neighbors-1} g_l^{2T_neighbors} * ( f(i+l) - f(i-1-l) ) / dx,
+     * where D_x is the derivative operator along x, dx the grid spacing along x, and
+     * g_l^{2T_neighbors} weightings for the finite differences of different distance l from the point i
+     * of computation.
+     *
+     * @tparam T_neighbors Number of neighbors used to calculate
+     *                     the derivative from finite differences.
+     *                     Order of derivative approximation is
+     *                     2 * T_neighbors
+     *
+     * @tparam T_direction direction to take derivative in, 0 = x, 1 = y, 2 = z
+     */
+    template<
+        uint32_t T_neighbors,
+        uint32_t T_direction
+    >
+    using BackwardDerivativeFunctor = detail::GeneralAofdtdDerivative< T_neighbors, T_neighbors - 1, T_neighbors, T_direction >;
+
+} // namespace aoFDTD
+} // namespace maxwellSolver
+
+namespace differentiation
+{
+namespace traits
+{
+
+    /** DerivativeFunctor type trait specialization for Forward derivative in
+     *  ArbitraryOrderFDTD solver
+     *
+     * @tparam T_neighbors Number of neighbors used to calculate
+     *                      the derivative from finite differences.
+     *                      Order of derivative approximation is
+     *                      2 * T_neighbors
+     *
+     * @tparam T_direction direction to take derivative in, 0 = x, 1 = y, 2 = z
+     */
+     template<
+        uint32_t T_neighbors,
+        uint32_t T_direction
+    >
+    struct DerivativeFunctor<
+        maxwellSolver::aoFDTD::Forward< T_neighbors >,
+        T_direction
+    > : pmacc::meta::accessors::Identity<
+        maxwellSolver::aoFDTD::ForwardDerivativeFunctor<
+            T_neighbors,
+            T_direction
+        >
+    >
+    {
+    };
+
+
+    /** DerivativeFunctor type trait specialization for Backward derivative in
```

Comment:
Please use doxygen groups https://www.doxygen.nl/manual/grouping.html
You can also have a bried description per struct.